examples/model_selection/plot_successive_halving_iterations.py

```diff
@@ -0,0 +1,72 @@
+"""
+Successive Halving Iterations
+=============================
+
+This example illustrates how
+:class:`HalvingGridSearchCV <sklearn.model_selection.HalvingGridSearchCV>`
+and
+:class:`HalvingRandomSearchCV <sklearn.model_selection.HalvingRandomSearchCV>`
+selectively choose the best parameter combination out of multiple candidates.
+
+At the first iteration, a small amount of resources is used. The resource here
+is the number of samples that the estimators are trained on. All candidates are
+evaluated.
+
+At the second iteration, only the best half of the candidates is evaluated.
+The number of allocated resources is doubled: candidates are evaluated on
+twice as many samples.
+
+This process is repeated until the last iteration, where only 2 candidates
+are left. The best candidate is the candidate that has the best score at the
+last iteration.
```

Comment:
we don't, but that's still the ideal case

Reply:
I don't understand why. I don't think your arguments here are convincing.

Reply:
https://github.com/scikit-learn/scikit-learn/pull/13900#discussion_r318068365

> why we need to have exactly ratio number of candidates on the last iteration.

For clarification, we don't need *exactly* `ratio` candidates, we need `<= ratio` candidates.

I'll try to clarify that part on the UG (marking this as resolved since this can be better tracked on the UG comments)